In [153]:
russian_dictionary = {
            'А': 10, 'Б': 11, 'В': 12, 'Г': 13, 'Д': 14, 'Е': 15, 'Ж': 16,
            'З': 17, 'И': 18, 'Й': 19, 'К': 20, 'Л': 21, 'М': 22, 'Н': 23,
            'О': 24, 'П': 25, 'Р': 26, 'С': 27, 'Т': 28, 'У': 29, 'Ф': 30,
            'Х': 31, 'Ц': 32, 'Ч': 33, 'Ш': 34, 'Щ': 35, 'Ъ': 36, 'Ы': 37,
            'Ь': 38, 'Э': 39, 'Ю': 40, 'Я': 41
        }

english_dictionary = {
            'A': 42, 'B': 43, 'C': 44, 'D': 45, 'E': 46, 'F': 47, 'G': 48,
            'H': 49, 'I': 50, 'J': 51, 'K': 52, 'L': 53, 'M': 54, 'N': 55,
            'O': 56, 'P': 57, 'Q': 58, 'R': 59, 'S': 60, 'T': 61, 'U': 62,
            'V': 63, 'W': 64, 'X': 65, 'Y': 66, 'Z': 67
        }

special_symbols_encoding = {
            ' ': 68, ',': 69, '.': 70
        }

def convert_message(message:str):
    message_in_coding = ''

    for letter in message:
        if letter in russian_dictionary:
            message_in_coding += str(russian_dictionary[letter])
        elif letter in english_dictionary:
            message_in_coding += str(english_dictionary[letter])
        elif letter in special_symbols_encoding:
            message_in_coding += str(special_symbols_encoding[letter])

    return message_in_coding

In [154]:
def create_number_for_encryption(number:str, key:int):
    number_for_encryption = []
    step = 0
    for i in range(1, len(number)):
        if int(number[step:i]) < key and int(number[step:(i + 1)]) > key:
            if int(number[i]) == 0:
                number_for_encryption.append(number[step:i - 1])
                step = i - 2
                i -=1
            else:
                number_for_encryption.append(number[step:i])
                step = i
        elif i + 2 >= len(number):
            number_for_encryption.append(number[step:])
            break

    return number_for_encryption  

In [156]:
number = convert_message('Hello world'.upper())
print(number)

key = 21841

print(create_number_for_encryption(number, key))

4946535356686456595345
['4946', '5353', '5668', '6456', '5953', '45']


21841

4946 5353 5668 6456 5953 45

    for i in range(1, len(number)):
        try:
            if int(number[step:i]) < key and int(number[step:(i + 1)]) > key:
                if int(number[i]) == 0:
                    number_for_encryption.append(number[step:i - 1])
                    step = i - 2
                    i -=1
                else:
                    number_for_encryption.append(number[step:i])
                    step = i
        except IndexError:
            number_for_encryption.append(number[step:])

    return number_for_encryption  